src/controllers/accounts.js

```diff
@@ -112,8 +112,10 @@ function * putResource () {
   const name = this.params.name
   request.validateUriParameter('name', name, 'Identifier')
   const account = this.body
-  request.assert.strictEqual(account.name, name.toLowerCase(),
-    'Account name must match the one in the URL')
+  if (account.name) {
+    request.assert.strictEqual(account.name.toLowerCase(), name.toLowerCase(),
```

Comment:
This was to make the tests pass, I'm not sure exactly what we want here, so I went with what the tests were expecting.